src/core/borrower/shelf.sol

```diff
@@ -181,7 +179,7 @@ contract Shelf is DSNote, Auth, TitleOwned, Math {
     }
 
     // --- NFT actions ---
-    function lock(uint loan, address usr) public owner(loan) {
+    function lock(uint loan) public owner(loan) {
```

Comment:
lets say A owns the NFT and the loan. (and he/she added the approve call)

A sends NFT to B. If B would give an approve.

A could still lock the NFT with this method. Correct?

If B would create a new loan with the NFT. Could A still call this method? Would the registry entry still exist?

Reply:
Let's maybe to avoid all the edge case always require the following

```javascript
require(msg.sender == title.ownerOf(loan) == NFTLike(shelf[loan].registry).ownerOf(shelf[loan].tokenId))
```

Reply:
I think it is already given, because the msg.sender has to be the nftOwner and the loanOwner for lock to work.
NFTLike(shelf[loan].registry).transferFrom(msg.sender, address(this), shelf[loan].tokenId); -> makes sure user is nftOwner
modifier: owner(loan) -> makes sure user is loanOwner